compiler/src/dotty/tools/dotc/ast/DesugarEnums.scala

```diff
@@ -126,19 +126,19 @@ object DesugarEnums {
    *
    *   private def $new(_$ordinal: Int, $name: String) = new E with scala.runtime.EnumValue {
    *     def $ordinal = $tag
-   *     override def toString = $name
+   *     override def productPrefix = $name     // if does not derive from `java.lang.Enum`
```

Comment:
I suggest being a bit more explicit in comments, e.g.:
```suggestion
   *     override def productPrefix = $name     // if this case does not derive from `java.lang.Enum`
```

Reply:
I've changed to ``if `E` does not derive from `java.lang.Enum` ``